### ISSUE TITLE
'format date' showing incorrect date format

### ISSUE THREAD

**Issue by dmatos2012** [nushell/nushell#10838]
### Describe the bug

Hi,
Formatting date using the date format '%x' does not format date as it would be expected. It looks that the code uses `sys-locale` crate which takes the first entry on `locale` which may not be desired.  At least on my platform, first entry is `LANG`, but I would expect `date` to be formatted with `LC_TIME` or `LC_ALL` maybe?. 

For comparison, using built-in `date` 

### How to reproduce

```
>date now | format date %x
10/25/23
```

### Expected behavior

```
>date now | format date %x
25/10/23.
```
For comparison, using the built-in `date` in Linux, yields:
```
^date --date='now' +%x`
25-10-23
```
Which I guess is using my `LC_TIME` locale. 
```
> locale
LANG=en_US.UTF-8
...
LC_TIME=nl_NL.UTF-8
..
```

             

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.86.1                                   |
| branch             | uu_mv                                    |
| commit_hash        | fbe88485eb1c45f8442fb5ca20b39aec219d5b80 |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.71.1 (eb26296b5 2023-08-03)      |
| rust_channel       | 1.71.1-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.71.1 (7f1d04c00 2023-07-29)      |
| build_time         | 2023-10-25 14:57:18 +02:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  |                                          |

**Comment by fdncred** [comment]
Maybe we should look for LC_TIME first then fallback to LANG if it's not found?

**Comment by stellarpower** [comment]
I just noticed this on my prompt. There aren't 28 months in the year!!! Assume this fix here would also affect the prompt output for default config, but just highlighting as a possible test case.

**Comment by stellarpower** [comment]
IDK if this is related - but `19:52:18 PM` is what my prompt shows right now, and this doesn't technically make much sense, to combine 24-hour with "PM"

**Comment by fdncred** [comment]
yes, of course. I made that change because it seems the most intuitive without adding unnecessary complexity to the prompt. Feel free to submit a PR.

**Comment by dmatos2012** [comment]
I'm working on it , just writing some tests and then I'll submit a PR :)

**Comment by fdncred** [comment]
Thanks David!

**Comment by dmatos2012** [comment]
Actually, not so fast :stuck_out_tongue: . I thought i was "finished", but then I realized while all curent tests passed, creating new ones showed some things were fixed, and others werent.  Particularly,

```
use chrono::prelude::*;
fn main() {
    let locale: Locale = Locale::nl_NL; # switch this
    let formatter = "%X";
    let ime: DateTime<Local> = Local::now(); 
    dbg!(&local_time);
    let format = time.format_localized(formatter, locale);
    println!("{}", format);
}

```
```
> cargo r
20:22
```

Then switching locale for `Locale::en_US
```
> cargo r
08
```

But If i format the date rather than time, it works fine. So same script as above, but with %x instead
Locale::en_US
```
> cargo r
10/29/23
```

Locale::nl_NL
```
> cargo r
29-10-23. 
```

And according to documentation, both are Locale's date/time representation. I did find from this [chrono comment](https://github.com/chronotope/chrono/issues/708#issuecomment-1154921227), that there are many conditions and what locale/ or not are used and which way is working for different platforms. 

As far as I could see, the `%p` formatter which is also locale worked fine, so not sure whether the time representation is a bug or intended. 

 I never knew time formatting/handling could be  this tricky. Also in the future this could be new candidate for `uutils` date maybe?

For example
```
^date '+%x +%X #uses my LC_TIME = nl_NL locale.
29-10-23 +20:34:02
```
```
>$env.LC_TIME = "en_US.UTF-8"
>^date '+%x +%X
10/29/2023 +08:34:32 PM
```
TLDR: IS time representation a bug or intended? Current fix that I have fixes:
* The commenter issue here with time (19:34PM) 
* Date representation using wrong locale.

However, if you are in `Locale::en_US` then the time representation will be broken for you:D

**Comment by stellarpower** [comment]
> 08
That's as expected, it's just doing the time in pound-inches of mercury per Lambert and then rounding for you :p

**Comment by develop7** [comment]
Affects me too, the locale in question is `be_BY.UTF-8`

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.88.1                                                            |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.74.0 (79e9716c9 2023-11-13) (built from a source tarball) |
| cargo_version      | cargo 1.74.0                                                      |
| build_time         | 2023-12-18 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | default, extra, sqlite, trash, which, zip                         |
| installed_plugins  |                                                                   |
